src/structurizr/view/dynamic_view.py

```diff
@@ -0,0 +1,317 @@
+from ..model.static_structure_element import StaticStructureElement
+from .relationship_view import RelationshipView
+from .sequence_number import SequenceNumber
+from .view import View, ViewIO
+
+
+__all__ = ("DynamicView", "DynamicViewIO")
+
+
+class DynamicViewIO(ViewIO):
+    """
+    Represent the dynamic view from the C4 model.
+
+    Attributes:
+        element: The software system or container that this view is focused on.
+    """
+
+    element_id: Optional[str] = Field(default=None, alias="elementId")
+
+
+class DynamicView(ModelRefMixin, View):
+    """
+    Represent the dynamic view from the C4 model.
+
+    A dynamic diagram can be useful when you want to show how elements in a static
+    model collaborate at runtime to implement a user story, use case, feature, etc.
+    This dynamic diagram is based upon a UML communication diagram (previously known
+    as a "UML collaboration diagram"). It is similar to a UML sequence diagram
+    although it allows a free-form arrangement of diagram elements with numbered
+    interactions to indicate ordering.
+
+    Attributes:
+        element: The software system or container that this view is focused on.
+    """
+
+    def __init__(
+        self,
+        *,
+        software_system: Optional[SoftwareSystem] = None,
+        container: Optional[Container] = None,
+        **kwargs,
+    ) -> None:
+        """Initialize a DynamicView.
+
+        Note that we explicitly don't pass the software_system to the superclass as we
+        don't want it to appear in the JSON output (DynamicView uses elementId
+        instead).
+        """
+        if software_system is not None and container is not None:
+            raise ValueError("You cannot specify both software_system and container")
+        super().__init__(**kwargs)
+        self.element = software_system or container
+        self.element_id = self.element.id if self.element else None
+        self.sequence_number = SequenceNumber()
+
+    def add(
+        self,
+        source: Element,
+        destination: Element,
+        description: Optional[str] = None,
+        *,
+        technology: Optional[str] = None,
+    ) -> RelationshipView:
+        """Add a relationship to this DynamicView.
+
+        This will search for a relationship in the model from the source to the
+        destination with matching technology (if specified).  It will also look for
+        situations where this interaction is a "response" in that it it goes in the
+        opposite direction to the relationship in the model.  If a description is
+        provided then this will be used in the view in preference to the description
+        on the relationship.
+
+        Example of request/response, assuming a single relationship in the model:
+            dynamic_view.add(container1, container2, "Requests data from")
+            dynamic_view.add(container2, container1, "Sends response back to")
```

Comment:
Let's use the [Google-style docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) sections.

```suggestion
        Examples:
            Example of a request/response, assuming a single relationship in the model:
            
                dynamic_view.add(container1, container2, "Requests data from")
                dynamic_view.add(container2, container1, "Sends response back to")

```